CourseSearchAPIHttpTrigger/query.py

```diff
@@ -0,0 +1,185 @@
+import logging
+import os
+
+
+def build_course_search_query(
+    course, institution, institutions, postcode_object, query_params
+):
+
+    try:
+        query = Query(course, institution, institutions, postcode_object, query_params)
+
+        query.build()
+
+        return query.add_paging()
+    except Exception:
+        raise
+
+
+def build_institution_search_query(
+    course, institution, institutions, postcode_object, query_params
+):
+
+    try:
+        query = Query(course, institution, institutions, postcode_object, query_params)
+
+        query.build()
+
+        return query.add_facet()
+    except Exception:
+        raise
+
+
+class Query:
+    def __init__(
+        self, course, institution, institutions, postcode_object, query_params
+    ):
+
+        self.postcode_object = postcode_object
+        self.course = course
+        self.institution = institution
+        self.institutions = institutions
+        self.query_params = query_params
+
+    def build(self):
+        query = ""
+        # Create search part of query
+        search = list()
+        if self.institution != "":
```

Comment:
Could be written more simply as:

if self.institution:
    search.append(self.institution)